app/src/androidTest/java/com/github/se/cyrcle/ui/gambling/GamblingScreenTest.kt

```diff
@@ -158,7 +199,25 @@ class GamblingScreenTest {
     composeTestRule.mainClock.autoAdvance = false
 
     composeTestRule.onNodeWithTag("spin_button").performClick()
-    composeTestRule.mainClock.advanceTimeBy(10000)
+    // Wait for second spin to complete
+    run {
+      val timeoutMs = 15000L
+      val incrementMs = 1000L
+      var elapsedTime = 0L
+
+      while (elapsedTime < timeoutMs) {
+        composeTestRule.mainClock.advanceTimeBy(incrementMs)
+        Thread.sleep(1000)
```

Comment:
Does the Thread.sleep need to be here?

Reply:
yes it does

Reply:
Why is that?

Reply:
Good question, i dont have the answer but the test doesnt pass without, or with a lower value...

Reply:
Ok got it, I'll let it go as-is then.